onnx_trt_backend.cpp

```diff
@@ -349,29 +349,45 @@ onnxStatus GraphRep::Run() {
 
 } // namespace
 
+#define ONNXIFI_CATCH_EXCPETION()                                              \
+  catch (const std::exception &e) {                                            \
```

Comment:
I suggest to separately catch `std::bad_alloc` and return `ONNXIFI_STATUS_NO_SYSTEM_MEMORY` in this case.